File: src/main/java/net/kigawa/util/plugin/player/PlayerGetter.java
package net.kigawa.util.plugin.player;

import net.kigawa.util.plugin.plugin.KigawaPlugin;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public class PlayerGetter {
    KigawaPlugin plugin;
    public PlayerGetter(KigawaPlugin kigawaPlugin){
        plugin=kigawaPlugin;
    }
    public List<Player> room(String world,int x,int y,int z,int x1,int y1,int z1){
        int sx;
        int bx;
        int sy;
        int by;
        int sz;
        int bz;
        if (x>x1){
            sx=x1;
            bx=x+1;
        }else{
            sx=x;
            bx=x1+1;
        }
        if (x>x1){
            sy=y1;
            by=y+1;
        }else{
            sy=y;
            by=y1+1;
        }
        if (x>x1){
            sz=z1;
            bz=z+1;
        }else{
            sz=z;
            bz=z1+1;
        }
        World world1=plugin.getServer().getWorld(world);
        Location location;
        List<Player> allPlayers=world1.getPlayers();
        List<Player> inRoom=new ArrayList<>();
        Player player;
        boolean isIn;
        for (int i = 0;i<allPlayers.size();i++){
            isIn=false;
            player=allPlayers.get(i);
            location=player.getLocation();
            if (location.getX()>sx&&location.getY()>sy&&location.getZ()>sz){
                isIn=location.getX()<bx&&location.getY()<by&&location.getZ()<bz;
            }
            if (isIn){
                inRoom.add(player);
            }
        }
        return inRoom;
    }
}

File: src/main/java/net/kigawa/onigoplugin/util/plugin/all/player/PlayerGetter.kt
package net.kigawa.onigoplugin.util.plugin.all.player

import net.kigawa.kutil.unitapi.annotation.Kunit
import net.kigawa.onigoplugin.OnigoPlugin
import org.bukkit.Location
import org.bukkit.entity.Player

@Kunit
class PlayerGetter(var plugin: OnigoPlugin) {
  fun room(world: String, x: Int, y: Int, z: Int, x1: Int, y1: Int, z1: Int): MutableList<Player> {
    var sx: Int
    var bx: Int
    var sy: Int
    var by: Int
    var sz: Int
    var bz: Int
    if (x > x1) {
      sx = x1
      bx = x
    } else {
      sx = x
      bx = x1
    }
    if (y > y1) {
      sy = y1
      by = y
    } else {
      sy = y
      by = y1
    }
    if (z > z1) {
      sz = z1
      bz = z
    } else {
      sz = z
      bz = z1
    }
    if (sx < 0) sx--
    if (sy < 0) sy--
    if (sz < 0) sz--
    if (bx > 0) bx++
    if (by > 0) by++
    if (bz > 0) bz++
    val world1 = plugin.server.getWorld(world)
    var location: Location
    val allPlayers = world1!!.players
    plugin.logger("all player " + allPlayers.size)
    plugin.logger("sx $sx")
    plugin.logger("bx $bx")
    val inRoom: MutableList<Player> = ArrayList()
    var player: Player
    var isIn: Boolean
    for (i in allPlayers.indices) {
      isIn = false
      player = allPlayers[i]
      plugin.logger(player.name)
      location = player.location
      plugin.logger("x" + location.x)
      plugin.logger((location.x >= sx).toString() + (location.y >= sy) + (location.z >= sz))
      if (location.x >= sx && location.y >= sy && location.z >= sz) {
        plugin.logger("check big")
        isIn = location.x <= bx && location.y <= by && location.z <= bz
      }
      if (isIn) {
        inRoom.add(player)
      }
    }
    plugin.logger("in room " + inRoom.size)
    return inRoom
  }
}
